tests/director/tests/formation_api_test.go

```diff
@@ -4252,6 +4254,304 @@ func TestFormationNotificationResynchronizationAsync(stdT *testing.T) {
+		assertFormationStatus(t, ctx, subscriptionConsumerAccountID, formation.ID, graphql.FormationStatus{Condition: graphql.FormationStatusConditionReady, Errors: nil})
+
+		body = getNotificationsFromExternalSvcMock(t, certSecuredHTTPClient)
+		assertFormationNotificationFromCreationOrDeletion(t, body, formation.ID, formation.Name, createFormationOperation, subscriptionConsumerAccountID, "")
+
+		cleanupNotificationsFromExternalSvcMock(t, certSecuredHTTPClient)
+
+		t.Logf("Assign tenant %s to formation %s", subscriptionConsumerSubaccountID, formationName)
+		defer fixtures.CleanupFormationWithTenantObjectType(t, ctx, certSecuredGraphQLClient, formationName, subscriptionConsumerSubaccountID, subscriptionConsumerAccountID)
+		assignedFormation := fixtures.AssignFormationWithTenantObjectType(t, ctx, certSecuredGraphQLClient, graphql.FormationInput{Name: formationName}, subscriptionConsumerSubaccountID, subscriptionConsumerAccountID)
+		require.Equal(t, "INITIAL", assignedFormation.State)
+
+		expectedAssignments := map[string]map[string]fixtures.AssignmentState{
+			runtimeContextID: {runtimeContextID: fixtures.AssignmentState{State: "READY", Config: nil}},
+		}
+		assertFormationAssignments(t, ctx, subscriptionConsumerAccountID, formation.ID, 1, expectedAssignments)
+		// Should be InProgress when status API is changed to reflect formation state
+		assertFormationStatus(t, ctx, subscriptionConsumerAccountID, formation.ID, graphql.FormationStatus{Condition: graphql.FormationStatusConditionReady, Errors: nil})
+
+		assertNoNotificationsAreSent(t, certSecuredHTTPClient)
+
+		t.Logf("Assign application to formation %s", formation.Name)
+		defer fixtures.CleanupFormation(t, ctx, certSecuredGraphQLClient, graphql.FormationInput{Name: formationName}, actualApp.ID, graphql.FormationObjectTypeApplication, subscriptionConsumerAccountID)
+		assignedFormation = fixtures.AssignFormationWithApplicationObjectType(t, ctx, certSecuredGraphQLClient, graphql.FormationInput{Name: formationName}, actualApp.ID, subscriptionConsumerAccountID)
+
+		expectedAssignments = map[string]map[string]fixtures.AssignmentState{
+			runtimeContextID: {
+				runtimeContextID: fixtures.AssignmentState{State: "READY", Config: nil},
+				actualApp.ID:     fixtures.AssignmentState{State: "INITIAL", Config: nil},
+			},
+			actualApp.ID: {
+				actualApp.ID:     fixtures.AssignmentState{State: "READY", Config: nil},
+				runtimeContextID: fixtures.AssignmentState{State: "INITIAL", Config: nil},
+			},
+		}
+		assertFormationAssignments(t, ctx, subscriptionConsumerAccountID, formation.ID, 4, expectedAssignments)
+		assertFormationStatus(t, ctx, subscriptionConsumerAccountID, formation.ID, graphql.FormationStatus{Condition: graphql.FormationStatusConditionInProgress, Errors: nil})
+		assertFormationAssignmentsAsynchronously(t, ctx, subscriptionConsumerAccountID, formation.ID, 4, expectedAssignments)
+
+		assertNoNotificationsAreSent(t, certSecuredHTTPClient)
```

Comment:
Why no notifications are sent? both runtime/rtm ctx and application have webhooks?

Reply:
Because the formation is in initial state and no notifications should be sent. They should be sent only when the formation changes to ready state, currently there is nothing to change the state to READY.

Reply:
I get it, I guess that's part of the "workaround" before we have the status API logic? And after that we need to adjust these are assert notifications are sent?

Reply:
Maybe not adjust them, but add more after them - create formation and assign objects - check that they're in initial state, wait the duration for the status API to respond to the formation lifecycle notification and trigger the notifications, we should then check that the synchronous notifications are sent and states are updated. After waiting for the formation assignment status API to respond everything should be in ready state.